### ISSUE TITLE
Move dependencies to peer dependencies?

### ISSUE THREAD

**Issue by bradtaniguchi** [JLuboff/connect-mssql-v2#17]
I was recently looking at the `package.json` and realized this package has 2 dependencies:
```
"@types/mssql": "^6.0.0",
"mssql": "^6.1.0"
```

I looked into this to make sure this is the right way to do things, and ran into [this](https://stackoverflow.com/questions/54926369/what-dependency-type-should-types-be-for-a-published-package) SO post where it seems like using `peerDependencies` would be better (at least for the type def).

Looking further I looked into one of the more popular express-session stores, connect-session-sequelize's [package.json](https://github.com/mweibel/connect-session-sequelize/blob/master/package.json) and found the "sequalize dependency" under peerDependencies.

As such it might make more sense to move these under peerDependencies instead of dependencies, as this package falls under the "plugins" type I keep hearing mentioned for use cases for peerDependencies.

Further reading:
- [differences between the different dependencies in package.json from a SO post](https://stackoverflow.com/questions/18875674/whats-the-difference-between-dependencies-devdependencies-and-peerdependencies)
- [nodejs docs on peer dependencies](https://nodejs.org/es/blog/npm/peer-dependencies/)

**Comment by JLuboff** [comment]
@bradtaniguchi Sorry, just saw this right now. After reading what you posted and the links, I agree that  `peerDependencies` should be used. Now, if I'm understanding correctly, it is suggested to have both `@types/mssql` and `mssql` in `peerDependencies`?

**Comment by bradtaniguchi** [comment]
Yes, we should move those to peerDependencies. I **think** this might be a "breaking change", I'm not 100% sure about the `@types/mssql` though since that one is "optional" right? Not sure how to have an optional peer dependency ya know?

(busy ATM so can't look much into it)

**Comment by JLuboff** [comment]
I think you're right about having `@types/mssql` as a `devDependency`, however, I'll have to test as you said. What do you think is the best approach to that? Create a new branch, modify `package.json` , build the package locally and test within an existing project using `npm link`?

**Comment by bradtaniguchi** [comment]
Yea that sounds about right :) 

I've been setting up a mssql installation on my machine to help test this out aswell

**Comment by JLuboff** [comment]
Just FYI... started a new branch (https://github.com/JLuboff/connect-mssql-v2/tree/issue17). Updated dependencies, moved `mssql` to `peerDependencies` and `@types/mssql` to `devDependencies`. Also updated ESLint config due to updated dependencies.

**Comment by JLuboff** [comment]
Alright just tested using `npm link` and it did not like having `mssql` as a `peerDependencies` but worked fine when it was a regular `dependencies`. Building the package never complained about having the types in devDeps nor did it complain in the test projects build.

**Comment by bradtaniguchi** [comment]
What was your project code for testing it out? Or what was the error when you had mssql as a peerDependency?

I'll try to check this out later today locally on my side too.

**Comment by JLuboff** [comment]
I just tested against one of my existing projects (uninstalled npm version, ran npm link connect-mssql-v2). The error was `mssql is missing` (or along those lines) and thus app wouldn't start. When I switched `mssql` back the issue was no more.
Also...looks like we may be getting those updated `express-session` typings here soon... holdyourwaffle was able to get it to pass testing but its currently awaiting review.
